src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -32,25 +32,31 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         final SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
-        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
-                .filter(h -> h.getValue().getHearingDateTime().isAfter(LocalDateTime.now()))
-                .distinct()
-                .findFirst();
-
         final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
 
         if (HearingRoute.GAPS.equals(sscsCaseData.getRegionalProcessingCenter().getHearingRoute())) {
             response.addError(POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS);
+            return response;
         }
 
+        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
+                .filter(h -> h.getValue().getStart().isAfter(LocalDateTime.now()))
+                .distinct()
+                .findFirst();
+
         optionalHearing.ifPresentOrElse(hearing -> setPostponementRequest(hearing, sscsCaseData),
                 () -> response.addError("There are no hearing to postpone"));
 
         return response;
     }
 
     private void setPostponementRequest(Hearing hearing, SscsCaseData sscsCaseData) {
-        sscsCaseData.getPostponementRequest().setPostponementRequestHearingDateAndTime(hearing.getValue().getHearingDateTime().toString());
+        if (HearingRoute.GAPS.equals(sscsCaseData.getSchedulingAndListingFields().getHearingRoute())) {
```

Comment:
There are no postponements of GAPS cases. This scenario should throw an error like above.

Reply:
There is a dedicated test checking that a postponement request from GAPS throws an error and it does indeed throw an error by virtue of how the "handle" method is implemented (line 39).

If anything, this "if" should not exist at all. I'll make that change.

Reply:
I aggree with Sean on this, there is a isSandL method in utilities I used it for my pr, so if(!isSandL(caseData) { addError("Case is not list assist")
there should also be a feature flag for SandL in the canHandle method